### PR TITLE
Add the metrics for embedded journal snapshot

### PR DESCRIPTION
**Pull request by LuQQiu**
Add metrics for better understanding the embedded journal snapshot behaviors

### CODE REVIEW

File: core/server/common/src/main/java/alluxio/master/journal/raft/SnapshotReplicationManager.java [hunk 1]
```diff
@@ -171,8 +174,11 @@ public SnapshotReplicationManager(RaftJournalSystem journalSystem,
       String address = String.valueOf(client.getAddress());
       SnapshotDownloader<DownloadSnapshotPRequest, DownloadSnapshotPResponse> observer =
           SnapshotDownloader.forFollower(mStorage, address);
+      Timer.Context ctx = MetricsSystem
+          .timer(MetricKey.MASTER_EMBEDDED_JOURNAL_SNAPSHOT_DOWNLOAD_TIMER.getName()).time();
       client.downloadSnapshot(observer);
       return observer.getFuture().thenApplyAsync((termIndex) -> {
+        ctx.close();
```
Comment (gpang) [inline core/server/common/src/main/java/alluxio/master/journal/raft/SnapshotReplicationManager.java:181]: Will this get closed twice, once here and once in `whenComplete`?

**Review by gpang — COMMENTED** [review]
@LuQQiu a minor question

File: core/server/common/src/main/java/alluxio/master/journal/raft/SnapshotReplicationManager.java (hunk 1 above)
Comment (LuQQiu) [reply core/server/common/src/main/java/alluxio/master/journal/raft/SnapshotReplicationManager.java:181]: Good catch, the timer got close() twice which is incorrect. Removed the close() in completeRequest. Thanks!

**Review by gpang — APPROVED** [review]
LGTM

Have you done a sanity check for these metrics?

**Comment by yuzhu** [comment]
@LuQQiu are these metrics ready to go? almost always good to have more metrics as long as they are not expensive..

**Comment by LuQQiu** [comment]
@yuzhu Yeah, i do a small empty lines fix and wait for the test build, i think it's ready to merge

**Comment by LuQQiu** [comment]
alluxio-bot, merge this please

**Pull request merged**